crates/client/devnet/src/lib.rs

```diff
@@ -325,6 +335,162 @@ mod tests {
+        let compiled_contract_class_hash =
+            Felt::from_hex("0x0138105ded3d2e4ea1939a0bc106fb80fd8774c9eb89c1890d4aeac88e6a1b27").unwrap();
+
+        let declare_txn: BroadcastedDeclareTransaction =
+            BroadcastedDeclareTransaction::V3(BroadcastedDeclareTransactionV3 {
+                sender_address: sender_address.address,
+                compiled_class_hash: compiled_contract_class_hash,
+                signature: vec![],
+                nonce: Felt::ZERO,
+                contract_class: Arc::new(flattened_class),
+                resource_bounds: ResourceBoundsMapping {
+                    l1_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                    l2_gas: ResourceBounds { max_amount: 60000, max_price_per_unit: 10000 },
+                },
+                tip: 0,
+                paymaster_data: vec![],
+                account_deployment_data: vec![],
+                nonce_data_availability_mode: DataAvailabilityMode::L1,
+                fee_data_availability_mode: DataAvailabilityMode::L1,
+                is_query: false,
+            });
+
+        let res = chain.sign_and_add_declare_tx(declare_txn, sender_address);
+
+        let calculated_class_hash = sierra_class.clone().class_hash().unwrap();
+
+        assert_eq!(res.class_hash, calculated_class_hash);
+
+        chain.block_production.set_current_pending_tick(1);
+        chain.block_production.on_pending_time_tick().unwrap();
+
+        let block = chain.backend.get_block(&BlockId::Tag(BlockTag::Pending)).unwrap().unwrap();
+
+        assert_eq!(block.inner.transactions.len(), 1);
+        assert_eq!(block.inner.receipts.len(), 1);
+        log::info!("receipt: {:?}", block.inner.receipts[0]);
+
+        let class_info =
+            chain.backend.get_class_info(&BlockId::Tag(BlockTag::Pending), &calculated_class_hash).unwrap().unwrap();
+        match class_info {
+            ClassInfo::Sierra(info) => {
+                assert_eq!(info.compiled_class_hash, compiled_contract_class_hash);
+            }
+            ClassInfo::Legacy(_) => {}
+        }
+
+        let TransactionReceipt::Declare(receipt) = block.inner.receipts[0].clone() else { unreachable!() };
+
+        assert_eq!(
+            receipt,
+            DeclareTransactionReceipt {
+                transaction_hash: res.transaction_hash,
+                messages_sent: vec![],
+                events: vec![Event {
+                    from_address: ERC20_STRK_CONTRACT_ADDRESS,
+                    keys: receipt.events[0].keys.clone(),
+                    data: receipt.events[0].data.clone(),
+                }],
+                execution_resources: receipt.execution_resources.clone(),
+                actual_fee: FeePayment { amount: receipt.actual_fee.amount, unit: PriceUnit::Fri },
+                execution_result: receipt.execution_result.clone(),
```

Comment:
with assert_matches! you can skip fields you don't want to check afaik. the `execution_result` should be hardcoded to check for success